### PR TITLE
simplify pd RPC client and add a pd worker

### PR DESCRIPTION
**Pull request by siddontang**
+ Add a pd worker for asynchronous communication with pd. Later store will tell sometimes to pd regularly, and this can't block the event loop when pd is down.
+ Simplify pd RPC client, no need thread support now. 

@ngaut @disksing @BusyJay

### CODE REVIEW

File: src/raftstore/store/worker/pd.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+use util::HandyRwLock;
+use util::worker::Runnable;
+use pd::PdClient;
+
+// Use an asynchronous thread to tell pd something.
+// TODO: remove enum_variant_names if we add other task later.
+#[allow(enum_variant_names)]
+pub enum Task {
+    AskChangePeer {
```
Comment (ngaut) [inline src/raftstore/store/worker/pd.rs]: To make it clear, We may add a type to AskChangePeer to indicate removing or adding peer.

**Comment by BusyJay** [comment]
Seems no tests?

File: src/raftstore/store/worker/mod.rs [hunk 2]
```diff
@@ -14,7 +14,9 @@
 mod snap;
 mod split_check;
 mod compact;
+mod pd;
 
 pub use self::snap::{Task as SnapTask, Runner as SnapRunner};
 pub use self::split_check::{Task as SplitCheckTask, Runner as SplitCheckRunner};
 pub use self::compact::{Task as CompactTask, Runner as CompactRunner};
+pub use self::pd::{Task as PdTask, Runner as PdRunner};
```
Comment (BusyJay) [inline src/raftstore/store/worker/mod.rs]: I think we should wrap `mio::TcpStream` as an `AsyncTcpClient` instead.

Comment (siddontang) [inline src/raftstore/store/worker/mod.rs]: ? 
Here no any TcpStream.

**Comment by siddontang** [comment]
Origin tests can all cover these, no need more.

**Comment by disksing** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**